### PR TITLE
changes to Metadb po_lines_locations.sql with comments added

### PR DESCRIPTION
**Pull request by natalyapik**
```
All queries:
- [ ] Release notes added or updated in CHANGES.md
- [ ] Query runs without errors
- [ ] Query output is correct
- [ ] Query logic is clear and well documented
- [ ] Query is readable and properly indented with spaces (not tabs)
- [ ] Table and column names are in all-lowercase
- [ ] Quotation marks are used only where necessary

Report queries:
- [ ] Query has complete user documentation
    - [ ] Purpose of report
    - [ ] Sample output
    - [ ] Query instructions

Derived tables:
- [ ] Query begins with user documentation in comment lines
- [ ] File name is listed in `runlist.txt` after dependencies
- [ ] All columns have indexes
- [ ] Table is vacuumed and analyzed
```

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ## 1.5.0
 
+* Additional columns added for po_lines_locations:
+  pol_location_name and pol_location source to accomodate data structure change with adding holding_id field for po lines,
+  Added comments to all columns for po_lines_locations.sql
 * A new derived table for LDP 1.x, `instance_administrative_notes`,
```
Comment (adoerrer) [inline CHANGES.md:6]: If you want to leave  the  updated holdings_administrative_notes.sql on comments in this PR. We need to have it mentioned in the Changes.md as well.

File: sql_metadb/derived_tables/po_lines_location.sql [hunk 2]
```diff
@@ -17,25 +18,51 @@ WITH ploc AS (
...
+         ELSE loc2.name
+    END AS pol_location_name,
+    CASE WHEN loc.name IS NOT NULL THEN 'pol_location'
+         WHEN loc2.name IS NOT NULL THEN 'pol_holding'
+         ELSE 'no_source'
+    END AS pol_location_source
 FROM
     ploc
-    LEFT JOIN folio_inventory.location__t AS loc ON loc.id = ploc.pol_loc_id;
+    LEFT JOIN folio_inventory.holdings_record__t AS hr ON ploc.pol_hold_id::uuid= hr.id
```
Comment (adoerrer) [inline sql_metadb/derived_tables/po_lines_location.sql:36]: Typo in `ploc.pol_hold_id` should be `ploc.pol_holding_id`

This prevents the query to run.

File: sql_metadb/derived_tables/po_lines_location.sql [hunk 3]
```diff
@@ -17,25 +18,51 @@ WITH ploc AS (
 )
 SELECT
     ploc.pol_id,
-    ploc.pol_loc_id,
     ploc.pol_loc_qty,
     ploc.pol_loc_qty_elec,
     ploc.pol_loc_qty_phys,
-    loc.name AS location_name
+CASE WHEN ploc.pol_loc_id IS NOT NULL THEN ploc.pol_loc_id
```
Comment (adoerrer) [inline sql_metadb/derived_tables/po_lines_location.sql:24]: Typo `ploc.pol_loc_id` should be  `ploc.pol_location_id`

File: sql_metadb/derived_tables/po_lines_location.sql [hunk 4]
```diff
@@ -17,25 +18,51 @@ WITH ploc AS (
...
+    END AS pol_location_name,
+    CASE WHEN loc.name IS NOT NULL THEN 'pol_location'
+         WHEN loc2.name IS NOT NULL THEN 'pol_holding'
+         ELSE 'no_source'
+    END AS pol_location_source
 FROM
     ploc
-    LEFT JOIN folio_inventory.location__t AS loc ON loc.id = ploc.pol_loc_id;
+    LEFT JOIN folio_inventory.holdings_record__t AS hr ON ploc.pol_hold_id::uuid= hr.id
+    LEFT JOIN folio_inventory.location__t AS loc ON loc.id = ploc.pol_loc_id
```
Comment (adoerrer) [inline sql_metadb/derived_tables/po_lines_location.sql:37]: Typo `ploc.pol_loc_id` should be `ploc.pol_location_id`

**Review by adoerrer — CHANGES_REQUESTED** [review]
There are some Typos in the field name which causes the query not to run. And there's is additional query which updates the comments which need to be mentioned in the Changes.md

All queries:
- [ ] Release notes added or updated in CHANGES.md
- [ ] Query runs without errors
- [x] Query output is correct
- [x] Query logic is clear and well documented
- [x] Query is readable and properly indented with spaces (not tabs)
- [x] Table and column names are in all-lowercase
- [x] Quotation marks are used only where necessary

Derived tables:
- [x] Query begins with user documentation in comment lines
- [x] File name is listed in `runlist.txt` after dependencies
- [x] All columns have indexes
- [x] Table is vacuumed and analyzed

**Comment by amzoss** [comment]
@natalyapik if you want to "remove" the holdings_administrative_notes.sql file from this PR, you could add a commit that edits that file back to the currently approved version of the query: https://github.com/folio-org/folio-analytics/blob/main/sql_metadb/derived_tables/holdings_administrative_notes.sql

**Comment by amzoss** [comment]
And, in case it's not clear, I'm waiting to review this query until previously mentioned issues are addressed.

**Pull request closed without merging**